### PR TITLE
feat: flatten Core/Plus nested response shape onto IPDetails

### PR DESCRIPTION
**Pull request by briandconnelly**
## Summary

The standard ipinfo endpoint (used today by `getHandlerAsync`) returns flat fields. The Core/Plus endpoints (`api.ipinfo.io/lookup`) return geolocation under a nested `geo` block and AS info under a nested `as` key — those would be silently dropped by the flat `IPDetails` schema (Pydantic's default is `extra="ignore"`).

This adds a small `_flatten_nested_response` helper that promotes nested keys to the top level before model construction:

- `geo` block flattened; `country_code` becomes `country` (ISO alpha-2), and the geo block's `country` (full name) becomes `country_name`
- `as` block (a Python keyword) renamed to `asn`
- Top-level keys win over nested on conflict (defensive)

## Scope notes

- **No-op today.** The current `getHandlerAsync` endpoint never returns `geo`/`as` keys, so existing behavior is unchanged. This is groundwork that lets a future tier-selector (e.g. `IPINFO_API_TIER=core`) work cleanly without further model changes.
- **`mobile` / `anonymous` (Plus-only) intentionally deferred.** Mapping their inner field names onto `CarrierDetails` / `PrivacyDetails` requires a real Plus API response to verify; guessing field names risks silent mis-mapping, which is worse than no mapping. Easy follow-up once a Plus token is available for testing.

## Test plan

- [x] `uv run pytest` — 87 passed (8 new), 96.28% coverage
- [x] `uv run ruff check . && uv run ruff format --check .` — clean
- [x] `uv run ty check` — clean
- [x] Pre-commit hooks pass

The 8 new tests in `TestFlattenNestedResponse` cover: flat-passthrough, geo flattening, `as`→`asn` rename, top-level-wins precedence, missing-key edge cases, non-dict input, and an end-to-end Core-shape → `IPDetails` construction.

### CODE REVIEW

File: src/mcp_server_ipinfo/ipinfo.py [hunk 1]
```diff
@@ -14,6 +14,44 @@ def _utc_timestamp() -> str:
     return datetime.now(timezone.utc).isoformat()
 
 
+def _flatten_nested_response(details: dict) -> dict:
+    """Flatten Core/Plus API nested response shape onto the flat IPDetails schema.
```
[the remark this answers is not in the document]
Comment (briandconnelly) [reply src/mcp_server_ipinfo/ipinfo.py:18]: Good catch — fixed in 49cfc58 by deleting the defensive branch. Per CLAUDE.md, validating internal-call invariants we already trust (`Details.all` is always a `dict`) is exactly the kind of error handling we don't want. The narrower contract matches the type annotation now.

File: src/mcp_server_ipinfo/ipinfo.py [hunk 2]
```diff
@@ -14,6 +14,44 @@ def _utc_timestamp() -> str:
...
+        return details
+
+    out = dict(details)
+
+    geo = out.pop("geo", None)
+    if isinstance(geo, dict):
+        geo = dict(geo)
+        if "country_code" in geo:
+            if "country" in geo:
+                geo.setdefault("country_name", geo.pop("country"))
```
[the remark this answers is not in the document]
Comment (briandconnelly) [reply src/mcp_server_ipinfo/ipinfo.py]: Confirmed — this was a real bug that would have surfaced as a Pydantic validation error against the `^[A-Z]{2}$` regex on `IPDetails.country`. Fixed in 49cfc58: I now always map `geo.country` (full name) → top-level `country_name` and `geo.country_code` → top-level `country` (alpha-2) independently, so the two halves can be present individually without crossing wires.

File: tests/test_models.py [hunk 3]
```diff
@@ -139,3 +140,127 @@ def test_optional_fields(self):
...
+
+    def test_missing_country_code_leaves_country_alone(self):
+        """If geo lacks country_code, don't reshape country fields (avoid bad mapping)."""
+        nested = {
+            "ip": "8.8.8.8",
+            "geo": {"city": "Mountain View", "country": "US"},
+        }
+        flat = _flatten_nested_response(nested)
+        assert flat["country"] == "US"
+        assert "country_name" not in flat
```
[the remark this answers is not in the document]
Comment (briandconnelly) [reply tests/test_models.py]: Right — the test was encoding the buggy behavior. Renamed and rewritten in 49cfc58 (`test_geo_country_is_full_name_when_country_code_missing`): when `country_code` is absent, `geo.country` is now treated as a full name and promoted to `country_name`, with no top-level `country` set.
